### DISCUSSION TITLE
Return value of a Visit* function?

### DISCUSSION THREAD

**Discussion by dduugg** [yoheimuta/protolint#375] [Q&A] [answered]
The `ENUM_NAMES_LOWER_SNAKE_CASE` example custom rule [returns](https://github.com/yoheimuta/protolint/blob/5011675/_example/plugin/customrules/enumNamesLowerSnakeCaseRule.go#L58) `false` from `VisitEnum`.

Other [rules](https://github.com/yoheimuta/protolint/blob/5011675/internal/addon/rules/messagesHaveCommentRule.go#L70) return `true` in their Visit* methods.

What is the meaning of the return value?

**Comment by yoheimuta** [comment, marked as answer]
@dduugg The return value determines if the parser will continue to delve deeper into the enum. You can see this in action at https://github.com/yoheimuta/go-protoparser/blob/e4248a911a58f09fabca602fb643a4839ad12cc4/parser/enum.go#L86-L88.

In this scenario, there's no need to examine the contents of the enum since the rule only concerns the enum's name.
